MainWindow.cpp

```diff
@@ -108,8 +203,25 @@ void MainWindow::openSubWindow(buffers::TreeNode *item) {
 
 void MainWindow::openFile(QString fName) {
   QFileInfo fileInfo(fName);
-  this->setWindowTitle(fileInfo.fileName() + " - ENIGMA");
-  project = gmx::LoadGMX(fName.toStdString());
+  const QString suffix = fileInfo.suffix();
+
+  if (suffix == "gm81" || suffix == "gmk" || suffix == "gm6" || suffix == "gmd") {

```

Comment:
Another example of "A lot of logic that has nothing to do with the main window." Isn't there already a project manager / file loader that this would fit nicely into?

Reply:
No, we don't have sort of `ProjectManager` yet but that if-else ladder could actually go into libEGM since emake has an equivalent if-else ladder. I think I just avoided it because I know fundies was bugging me about using boost for file and directory manipulation since pre-C++14/17 or w/e lacks things like getting the file extension. I did not want to add Boost to libEGM because then the IDE depends on it too, which doesn't make sense because Qt has its own facilities for not just directory and file manipulation, but also command line parsing.

Reply:
Well, all right; I would make this the first thing to go when this file starts creeping up on 500 lines, again.

Reply:
Absolutely, like I said, we'll likely figure out libEGM by that time anyway.